src/components/Pagination.vue

```diff
@@ -0,0 +1,223 @@
+		<b-button-group>
+			<b-button
+				v-for="page in pages"
+				:id="'tab'+page.index"
+				:key="page.index"
+				class="pagination-button"
+				:style="cssVars"
+				:pressed="selectedIndex == page.index"
+				@click="handleClick(page)"
+			>
+				{{ page.value }}
+			</b-button>
+		</b-button-group>
+	</div>
+</template>
+
+<script>
+
+export default {
+	props: {
+		perPage: {
+			type: Number,
+			default: 5,
+			description: `The maximum number of elements
+				for each page of the element being paged.`
+		},
+		totalRows: {
+			type: Number,
+			default: 1,
+			description: `The total amount of rows in the element
+				being paged.`,
+		},
+		activeBgColor: {
+			type: String,
+			default: '#545b62',
+			description: `The RGB color code for the background
+				of the active selection of the component.`,
+		},
+		activeTextColor: {
+			type: String,
+			default: '#fff',
+			description: `The RGB color code for the text
+				of the active selection of the component.`,
+		},
+		hoverBgColor: {
+			type: String,
+			default: '#545b62',
+			description: `The RGB color code for the background
+				of the hover selection of the component.`,
+		},
+		hoverTextColor: {
+			type: String,
+			default: '#fff',
+			description: `The RGB color code for the text
+				of the hover selection of the component.`,
+		},
+		borderRadius: {
+			type: Number,
+			default: 5,
+			description: `The value, in pixels, of the border
+				radius of the whole element.`,
+		},
+	},
+
+	data() {
+		return {
+			pressed: false,
+			selectedIndex: 0,
+			selectedValue: 1,
+			pages: [],
+			defaultIndex: 0,
+			defaultValue: 1,
+		};
+	},
+
+	computed: {
+		pageCount() {
+			return Math.ceil(this.totalRows / this.perPage);
+		},
+
+		cssVars() {
+			return {
+				'--active-bg-color': this.activeBgColor,
+				'--active-text-color': this.activeTextColor,
+				'--hover-bg-color': this.hoverBgColor,
+				'--hover-text-color': this.hoverTextColor,
+				'--border-radius': `${this.borderRadius}px`,
+			}
+		}
+	},
+
+	watch: {
+		totalRows() {
+			this.pages = this.handlePages();
+		},
+
+		perPage() {
+			this.pages = this.handlePages();
+		}
+	},
+
+	beforeMount() {
+		this.pages = this.handlePages();
+		this.selectedIndex = this.defaultIndex;
+		this.selectedValue = this.defaultValue;
+	},
+
+	methods: {
+		handleClick(page) {
+			if (page.value !== '...') {
+				this.selectedValue = page.value;
+				this.selectedIndex = page.index;
+				this.pages = this.handlePages();
+				this.pressed = true;
+				this.$emit('input', page.index + 1);
+			}
+		},
+
+		handlePages() {
+			const pages = [];
+			const pageNumber = (this.pageCount < 15) ? this.pageCount : 15;
+			for (let i = 0; i < pageNumber; i += 1) {
+				pages[i] = {
+					index: i,
+					value: i + 1,
+				};
+				if (pages[i].value === this.selectedValue) {
+					this.selectedIndex = pages[i].index;
+				}
+			}
+			if (this.pageCount > 15) {
+				if (this.selectedValue > this.pageCount - 8) {
```

Comment:
It helps reallocating the active selection in a case where there's a large amount of pages. I think this could be improved in a future refactoring of the code.